tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -1178,6 +1179,57 @@ TEST_F(ValidationTest, UavBarrierFail) {
 TEST_F(ValidationTest, UndefValueFail) {
   TestCheck(L"..\\CodeGenHLSL\\UndefValue.hlsl");
 }
+// verify that containers that are not valid DXIL do not
+// get assigned a hash.
+TEST_F(ValidationTest, ValidationFailNoHash) {
+  if (m_ver.SkipDxilVersion(1, 8))
+    return;
+  CComPtr<IDxcBlob> pProgram;
+  LPCSTR pSource = "float main(snorm float b : B) : SV_DEPTH \
```

Comment:
It would be useful to have a comment explaining how this shader is expected to fail validation but not compilation (due to missing diagnostics for reading uninitialized values).

It's generally best to start with IL rather than HLSL to reliably produce a validator failure, since we could theoretically fix the missing diagnostics and inadvertently invalidate this test.

Additionally, you could use a raw string instead of escaped newlines for better readability/editability.  Like:
```c++
LPCSTR pSource = R"(
    float main(snorm float b : B) : SV_DEPTH
    {
        float a;
        return b + a;
    }
)";
```